components/director/internal/domain/document/service.go

```diff
@@ -81,7 +106,34 @@ func (s *service) Create(ctx context.Context, applicationID string, in model.Doc
 
 	id := s.uidService.Generate()
 
-	document := in.ToDocument(id, tnt, applicationID)
+	document := in.ToDocument(id, tnt, &applicationID)
+	err = s.repo.Create(ctx, document)
+	if err != nil {
+		return "", errors.Wrap(err, "while creating Document")
+	}
+
+	if in.FetchRequest != nil {
+		generatedID := s.uidService.Generate()
+		fetchRequestID := &generatedID
+		fetchRequestModel := in.FetchRequest.ToFetchRequest(s.timestampGen(), *fetchRequestID, tnt, model.DocumentFetchRequestReference, id)
+		err := s.fetchRequestRepo.Create(ctx, fetchRequestModel)
+		if err != nil {
+			return "", errors.Wrapf(err, "while creating FetchRequest for Document %s", id)
+		}
+	}
+
+	return document.ID, nil
+}
+
+func (s *service) CreateToPackage(ctx context.Context, packageID string, in model.DocumentInput) (string, error) {
```

Comment:
What do you think about a name `CreateInPackage`? And also please add TODO comment that eventually we will replace current Create method with this logic.